bin/yaml/cpp.yaml

```diff
@@ -699,6 +699,36 @@ compilers:
           fetch:
             - https://registrationcenter-download.intel.com/akdlm/irc_nas/18703/l_fortran-compiler_p_{name}_offline.sh install.sh
           script: *intel-one-install-script
+    nvhpc-sdk:
+      type: script
+      check_stderr_on_stdout: true
+      check_exe: hpc_sdk/Linux_{arch}/{version}/compilers/bin/nvc++ --version
+      targets:
+        - arch: "x86_64"
+          version: "22.7"
+          script: &nvhpc-sdk-install-script |
```

Comment:
The YAML stuff is cool, but we support hierarchical settings - if you move this script up to the same level as `check_exe` and `type` then it applies to all its children (unless overridden!)

Reply:
Will merge and fix up

Reply:
Ah I see the thing from above that you copied from -- there's definitely opportunites for excellence here on my part :blush: